sapientml_core/templates/model_templates/model.py.jinja

```diff
@@ -22,21 +22,34 @@ model = {{ model_name }}({{ silent }}random_state=random_state_model, {{ params
 from sklearn.multioutput import MultiOutputRegressor
 
 model = MultiOutputRegressor(model)
+
 {% elif is_multioutput_classification %}
 from sklearn.multioutput import MultiOutputClassifier
 
 model = MultiOutputClassifier(model)
 {% endif %}
 {% set xgbclassifier = "XGBClassifier" %}
-{% if model_name == xgbclassifier %}
+{% if pipeline.task.target_columns|length == 1 %}
+
 from sklearn.preprocessing import LabelEncoder
 
-label_encoder = LabelEncoder()
-target_train = pd.DataFrame(label_encoder.fit_transform(target_train), columns=TARGET_COLUMNS)
-{% endif %}
-{% if pipeline.task.target_columns|length == 1 %}
+if target_train.select_dtypes(include=['object']).columns.any():
+        str_columns = target_train.select_dtypes(include=['object']).columns
+        label_encoder= LabelEncoder()
+        for col in str_columns:
+                target_train[col] = label_encoder.fit_transform(target_train[col])
+                target_test[col] = label_encoder.transform(target_test[col])
+
 model.fit(feature_train, target_train.values.ravel())
 {% else %}
+
+from sklearn.preprocessing import LabelEncoder
+if target_train.select_dtypes(include=['object']).columns.any():
+        str_columns = target_train.select_dtypes(include=['object']).columns
+        label_encoder= LabelEncoder()
```

Comment:
I think `label_encoder= LabelEncoder()` should be move to in the for loop next for the consistency to the training script. That is, 

```python
        label_encoders = {}
        for col in str_columns:
                label_encoder= LabelEncoder()
                target_train[col] = label_encoder.fit_transform(target_train[col])
                target_test[col] = label_encoder.transform(target_test[col])
                label_encoders[col] = label_encoder
```